utils/utils_interface_base.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019 Bluzelle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+#include <include/bluzelle.hpp>
+#include <string>
+#include <vector>
+#include <peers_beacon/peer_address.hpp>
+
+
+namespace bzn
+{
+    class utils_interface_base
```

Comment:
We do something similar in bzapi, for the same reason (mockable). It would be nice to consolidate the code. Even better, put all the ESR and HTTP stuff into a single library and share it.